### PR TITLE
[YUNIKORN-1282] Resource metrics are not tracked correctly for all queue types

### PR DESCRIPTION
**Pull request by ycr-oss**
### What is this PR for?
* Track resource usage in metrics for non-leaf queues
* Update resource metrics during config and node updates

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1282

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/queue.go [hunk 1]
```diff
@@ -269,12 +269,14 @@ func (sq *Queue) setResources(resource configs.Resources) error {
 	} else {
 		log.Logger().Debug("max resources setting ignored: cannot set zero max resources")
 	}
+	sq.updateMaxResourceMetrics()
```
Comment (wilfred-s) [inline pkg/scheduler/objects/queue.go:272]: I think we need this inside the `if` clause with the check for `StrictlyGreaterThanZero`. If there is no max set the object is a non nil pointer with no content.  We do not allow 0 values without at least 1 value larger than 0. That means this call would not result in any metrics being generated. The for loop inside the  `updateMaxResourceMetrics` never executes. It thus is a call to nowhere.

Same for guaranteed resources.

File: pkg/scheduler/objects/queue.go [hunk 2]
```diff
@@ -1017,6 +1019,8 @@ func (sq *Queue) SetMaxResource(max *resources.Resource) {
 		zap.String("current max", sq.maxResource.String()),
 		zap.String("new max", max.String()))
 	sq.maxResource = max.Clone()
+	sq.updateGuaranteedResourceMetrics()
```
Comment (wilfred-s) [inline pkg/scheduler/objects/queue.go:1022]: We only set max so we should not set the guaranteed metric.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: pkg/scheduler/objects/queue.go (hunk 2 above)
Comment (ycr-oss) [reply pkg/scheduler/objects/queue.go]: Good catch. Removed

File: pkg/scheduler/objects/queue.go (hunk 1 above)
Comment (ycr-oss) [reply pkg/scheduler/objects/queue.go]: I agree. Updated

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**